test/examples/ucp_hello_world.c

```diff
@@ -471,10 +482,13 @@ int main(int argc, char **argv)
             goto err_peer_addr;
         }
     }
-    close(oob_sock);
 
     ret = run_test(server, ucp_worker);
 
+    /* Make sure remote is disconnected before destroying local worker */
+    barrier(oob_sock);
```

Comment:
yes - we must keep ucp worker (or uct iface, respectively) alive before closing connections to it.
on TCP such limitation does not exist because the kernel maintains the socket after process exits. Killing ucp_worker is analogous to "pulling the cable" for TCP.